src/Sql/PostgreSQL/Functions/user_create.sql

```diff
@@ -0,0 +1,103 @@
+CREATE OR REPLACE FUNCTION user_create
+(
+    _id                                 uuid,
```

Comment:
Those are the function parameters, they would be the actual data when calling this function

Reply:
So is the underscore a keyword character or something? What is the actual name of the parameter here? Is it `id` or `_id`.?

Reply:
_id is the variable. It has to different from the column name, else postgreSQL complains about ambiguity , hence the leading underscore which is standard practice. 

Eg for a WHERE clause In TSQL it would be the same as
WHERE id = @id

So in plpgsql you would do 
WHERE id = _id
For the same result

The alternatives are to use actual column names or even no names and just use datatypes and use $1, $2 etc in the statements which is ugly

Using the function would be something like

SELECT * FROM user_case ('aaaa-bbbb....','....)

Or any number of other ways to call functions

Reply:
There is a difference though.

In TSQL, the `@` is a keyword character signifying a parameter. It's not part of the parameter name. It's a requirement. So the parameter `@Id` is actually named `Id`. If I am understanding correctly, in pgsql the leading underscore is just a convention. Here the actual name of the parameter here is `_id`.

The reason this is an issue is because of how Dapper maps the parameters. A dapper parameter from an object with the name "Id" is not going to map to a function parameter named `_id` since it is actually two different names.

There may be a way we can modify dapper to always prefix the underscore though.

Reply:
im not familiar with dapper, but why would it even reference the variable directly? 
is it using named notation rather than positional notation or something?

https://www.postgresql.org/docs/11/sql-syntax-calling-funcs.html

Reply:
If you pass a C# object to dapper, it transforms that object's property names to SQL parameters.

So in the case:

    public class User
    {
        public Guid Id { get; set; }
        public String Name { get; set; }
        // ...
    }

...

    public virtual async Task CreateAsync(User obj)
    {
        using(var connection = new NpgsqlConnection(ConnectionString))
        {
            var results = await connection.ExecuteAsync(
                "user_create",
                obj,
                commandType: CommandType.StoredProcedure);
        }
    }
    
 Dapper automatically maps the user object to function parameters (using reflection on the object property names).

Therefore our parameters in this example will end up being "Id" and "Name".

Reply:
ok, ill take your word for it, not familiar with dapper or C#. just let me know what variable names you want if this doesn't suit, but they can't be the same as the actual column names in this kind of function setup.

Reply:
I had some time to look into this more today. I was able to come up with a solution that will re-format the C# object properties to make the underscore-prefixed function parameters correctly. See https://github.com/bitwarden/server/commit/12c8e4b124fbeb7ba28bbddfcfb62266ad0a0058

Reply:
ok this is great!